### PR TITLE
GitHub Webhook change and a start on tests

### LINKED ISSUE #1: Add test cases for HTTP requests
We currently don't have any tests, as a start point, tests for the following are needed: 
- [ ] GitHub webhook
- [ ] Send message API request
- [x] Ping command (This may require a special use case)

### PR DESCRIPTION
**Pull request by BaileyJM02**
I'm currently in the process of adding tests to the project. This pull request is related to issue #1.

### CODE REVIEW

File: internal/pkg/github/github.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ type Webhook struct {
 	Release    Release    `json:"release"`
 	Repository Repository `json:"repository"`
 	Sender     Sender     `json:"sender"`
+	ChannelID  string     `json:""`
```
Comment (jeroenrinzema) [inline internal/pkg/github/github.go]: Do you want to include this property when marshalling to JSON? If not please use: `json:"-"`

File: internal/pkg/github/webhook.go [hunk 2]
```diff
@@ -16,6 +17,10 @@ func StartWatching() {
 // WebhookListener is the endpoint for which the GitHub webhook events should be sent
 // It checks if we support the sent event and handles it accordingly
 func WebhookListener(rw http.ResponseWriter, req *http.Request) {
+	if (strings.Split(req.URL.Path, "/")[2] == "") {
+		rw.WriteHeader(404) // Return 404 Not Found
+		return
+	}
```
Comment (jeroenrinzema) [inline internal/pkg/github/webhook.go]: Maybe using a small library would be easier for future use. We use [github.com/julienschmidt/httprouter](https://github.com/julienschmidt/httprouter) internally for most http routers.

**Review by jeroenrinzema — CHANGES_REQUESTED** [review]

File: internal/bot/bot.go [hunk 3]
```diff
@@ -24,9 +25,10 @@ func Start() {
...
 
 	// Add the endpoint for github webhook payloads
-	http.HandleFunc("/github", github.WebhookListener)
+	router := httprouter.New()
+	router.POST("/github/:channelID", github.WebhookListener)
 
 	// Start the HTTP server ()
 	address := os.Getenv("address")
-	logrus.Fatal(http.ListenAndServe(address, nil))
+	logrus.Fatal(http.ListenAndServe(address, router))
```
Comment (jeroenrinzema) [inline internal/bot/bot.go]: The default `ListenAndServe` method uses the default HTTP server. Some of the default values are set to extremely long values or are set to 0 (which disables them). Cloudflare has written an extensive blog post on how to set up a production-ready Go HTTP server and HTTP client. https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/

TL;DR:
```go
srv := &http.Server{
    ReadTimeout: 5 * time.Second,
    WriteTimeout: 10 * time.Second,
}

err := srv.ListenAndServe(address, router)
if err != nil && err != http.ErrServerClosed {
  logrus.Fatal(err)
}
```

**Review by jeroenrinzema — CHANGES_REQUESTED** [review]

File: internal/bot/bot.go (hunk 3 above)
Comment (BaileyJM02) [reply internal/bot/bot.go]: Finally got around to fixing this :sweat_smile:, thanks for the tip!

**Review by jeroenrinzema — APPROVED** [review]
